src/xGadgetron/cGadgetron/include/sirf/Gadgetron/gadgetron_data_containers.h

```diff
@@ -129,6 +115,67 @@ namespace sirf {
         mutable bool have_header_;
 	};
 
+	/*!
+	\ingroup Gadgetron Data Containers
+	\brief Class enabling ignoring certain 'irregular' acquisitions.
+	*/
+	class IgnoreMask {
+	public:
+		IgnoreMask(unsigned long int mask = (1 << 18)) :
+		    ignore_(mask), max_(8*sizeof(mask)) {}
+		void set(unsigned long int mask)
+		{
+			ignore_ = mask;
+		}
+		void ignore(unsigned int i)
+		{
+			if (i < 1 || i > max_)
+				return;
+			unsigned long int one = 1;
+			ignore_ = ignore_ | (one << (i - 1));
+		}
+		void ignore_not(unsigned int i)
+		{
+			if (i < 1 || i > max_)
+				return;
+			unsigned long int one = 1;
+			ignore_ = ignore_ & ~(one << (i - 1));
+		}
+		bool bit(unsigned int i) const
+		{
+			if (i < 1 || i > max_)
+				return true;
+			unsigned long int one = 1;
+			return ignore_ & (one << (i - 1));
+		}
+		unsigned long int bits() const
+		{
+			return ignore_;
+		}
+		bool ignored(unsigned long int bits) const
+		{
+			return bits & ignore_;
+		}
+		void show_bits() const
```

Comment:
I guess this is for debugging, but generally better to return a string. (can use `std::stringstream` to construct it), or pass a `ostream` as input

Reply:
done